src/checkers/inference/solver/backend/maxsat/MaxSatSolver.java

```diff
@@ -45,8 +47,10 @@
     }
 
     protected final SlotManager slotManager;
-    protected final List<VecInt> hardClauses = new LinkedList<VecInt>();
-    protected final List<VecInt> softClauses = new LinkedList<VecInt>();
+    protected final List<VecInt> hardClauses = new LinkedList<>();
+    private List<VecInt> wellFormdnessClauses = new LinkedList<>();
```

Comment:
`wellFormdnessClauses` is just for separation of concerns. Previously, well formdness constraints are also stored in `hardClauses` and caused some difficulties when we try to reversely lookup the original Constraint. Separating this makes explaning much easier.